src/typescript.js

```diff
@@ -15,37 +15,35 @@ module.exports = {
       },
       plugins: ['@typescript-eslint'],
       extends: [
+        'airbnb',
+        'airbnb-typescript',
         'plugin:import/typescript',
         'plugin:@typescript-eslint/eslint-recommended',
         'plugin:@typescript-eslint/recommended',
         'plugin:@typescript-eslint/recommended-requiring-type-checking',
+        'prettier',
       ],
       rules: {
-        // Define custom parcelLab rules
+        'no-param-reassign': ['error', { props: false }],
         '@typescript-eslint/naming-convention': [
           'error',
-          {
-            selector: 'default',
-            format: ['camelCase'],
-          },
-          {
-            selector: 'enumMember',
-            format: ['PascalCase'],
-          },
+          { selector: 'default', format: ['camelCase'] },
+          { selector: 'enumMember', format: ['PascalCase'] },
+          { selector: 'function', format: ['camelCase', 'PascalCase'] },
           {
             selector: 'parameter',
             format: ['camelCase'],
             leadingUnderscore: 'allowSingleOrDouble',
           },
+          { selector: 'typeLike', format: ['PascalCase'] },
+          { selector: 'variable', format: ['camelCase', 'UPPER_CASE'] },
           {
-            selector: 'typeLike',
-            format: ['PascalCase'],
-          },
-          {
-            selector: 'variable',
-            format: ['camelCase', 'UPPER_CASE'],
+            selector: ['typeProperty', 'objectLiteralProperty'],
+            format: null,
+            leadingUnderscore: 'allow',
           },
         ],
+        'no-unused-vars': 'off',
```

Comment:
In TS files, this rule is replaced by `@typescript-eslint/no-unused-vars`, to avoid conflicts the ESLint native `no-unused-vars` needs to be disabled.

Anyways, I removed it to avoid any misunderstanding. As it is by default removed by `@typescript-eslint`.
✅